src/common/playleaderboard/LeaderBoard.jsx

```diff
@@ -0,0 +1,106 @@
+  const [publishedPlays, updatePublishedPlays] = useState([]);
+  const { getAllPlaysByUser, getAllPlaysByUserByMonth } = FetchPlayCountByUser;
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const formatData = useCallback((data, monthlyContribution = false) => {
+    const formattedData = [];
+    const finalData = [];
+    const filteredData = data.filter((d) => publishedPlays.includes(d.slug));
+    filteredData.map((d) => {
+      formattedData.push({
+        created_at: d.created_at,
+        displayName: d.user.displayName,
+        avatarUrl: d.user.avatarUrl
+      });
+    });
+    const groupByUser = _.groupBy(formattedData, 'displayName');
+    Object.values(groupByUser).map((v) =>
+      finalData.push({
+        displayName: v[0].displayName,
+        avatarUrl: v.map((t) => t.avatarUrl)[0],
+        count: v.length,
+        created_at: v[0].created_at
+      })
+    );
+    const sortByCount = finalData.sort((a, b) => b.count - a.count);
+    if (monthlyContribution) {
+      // once the data being sorted based on created date, will pick the first one.
+      return sortByCount.sort((a, b) => new Date(b.created_at) - new Date(a.created_at))[0];
+    }
+
+    return sortByCount.slice(0, 10);
+  });
+
+  useEffect(() => {
+    async function fetchData() {
+      const res = await fetch(`${process.env.REACT_APP_PLAY_WEB_SVC}/api/plays/published`);
+      const data = await res.json();
+      updatePublishedPlays(data);
+    }
+    fetchData();
+  }, []);
+
+  useEffect(() => {
+    async function fetchTopContributors() {
+      const data = await submit(getAllPlaysByUser());
+      updateTop10Contributors(formatData(data));
+    }
+
+    async function fetchTopContributorOfTheMonth() {
+      const firstDateOfMonth = format(new Date(), 'yyyy-MM-01');
+      const lastDateOfMonth = format(lastDayOfMonth(new Date()), 'yyyy-MM-dd');
+      const data = await submit(getAllPlaysByUserByMonth(firstDateOfMonth, lastDateOfMonth));
+      updateTopContributorOfTheMonth(formatData(data, true));
+    }
+
+    fetchTopContributors();
+    fetchTopContributorOfTheMonth();
+  }, [publishedPlays]);
+
+  return (
+    <main className="app-body  app-body-overflow-hidden">
+      {publishedPlays.length && (topContributorOfTheMonth || top10Contributors) ? (
+        <div className=" overflow-auto lg:flex flex-row justify-around ">
+          {topContributorOfTheMonth && (
+            <TopPlayCreatorOfTheMonth topPlayCreatorOfTheMonth={topContributorOfTheMonth} />
+          )}
+          <div className="flex flex-col m-4 items-center">
+            {publishedPlays && top10Contributors && (
+              <>
+                <div className="heading">Top play creators of all time</div>
+                <div>
+                  <TopPlayCreators topPlayCreators={top10Contributors} />
+                </div>
+              </>
+            )}
+          </div>
+        </div>
+      ) : (
+        <div
+          style={{
```

Comment:
Can we take out the inline style to the CSS file?

Reply:
done